AdyenComponents/UPI/UPIComponent.swift

```diff
@@ -94,33 +108,80 @@ public final class UPIComponent: PaymentComponent,
-            self?.didChangeSegmentedControlIndex(changedIndex)
+        let item = FormSegmentedControlItem(
+            items: [
+                firstSegmentTitle,
+                localizedString(
+                    .UPISecondTabTitle,
+                    configuration.localizationParameters
+                )
+            ],
+            style: configuration.style.segmentedControlStyle,
+            identifier: ViewIdentifierBuilder.build(
+                scopeInstance: self,
+                postfix: ViewIdentifier.upiFlowSelectionItem
+            )
+        )
+        item.selectionHandler = { [weak self] in
+            self?.didChangeSegmentedControlIndex($0)
         }
         return item
     }()
 
     /// The  virtual payment address text input item.
-    internal lazy var virtualPaymentAddressItem: FormTextInputItem = {
+    internal lazy var vpaInputItem: FormTextInputItem = {
         let item = FormTextInputItem(style: configuration.style.textField)
-        item.title = Constants.virtualPaymentAddress
+        item.title = localizedString(
+            .UPICollectFieldLabel,
+            configuration.localizationParameters
+        )
         item.validator = LengthValidator(minimumLength: 1)
-        item.validationFailureMessage = localizedString(.UPIVpaValidationMessage, configuration.localizationParameters)
-        item.identifier = ViewIdentifierBuilder.build(scopeInstance: self,
-                                                      postfix: ViewIdentifier.virtualPaymentAddressInputItem)
+        item.validationFailureMessage = localizedString(
+            .UPIVpaValidationMessage,
+            configuration.localizationParameters
+        )
+        item.identifier = ViewIdentifierBuilder.build(
+            scopeInstance: self,
+            postfix: ViewIdentifier.virtualPaymentAddressInputItem
+        )
         return item
     }()
 
+    /// The UPI app list item.
+    internal lazy var upiAppsList: [SelectableFormItem] = {
+        guard let apps = upiPaymentMethod.apps, !apps.isEmpty else { return [] }
+        
+        var upiAppslist = apps.map { selectableFormItem(from: $0) }
+        upiAppslist.append(vpaSelectionItem)
+        return upiAppslist
+    }()
+
+    /// The UPI enter UPI/VPA list item.
+    internal lazy var vpaSelectionItem: SelectableFormItem = {
+        let selectableItem = SelectableFormItem(
+            title: localizedString(
+                .UPICollectDropdownLabel,
+                configuration.localizationParameters
+            ),
+            imageUrl: nil,
+            isSelected: false,
+            style: .init(title: configuration.style.textField.title),
+            identifier: Constants.vpaFlowIdentifier
+        )
+        selectableItem.selectionHandler = { [weak self, weak selectableItem] in
+            guard let self, let selectableItem else { return }
+            self.currentSelectedItemIdentifier = selectableItem.identifier
```

Comment:
Think we can combine the next 3 lines into the `updateSelection()` function.
If the identifier is vpa, we unhide it and hide the rest.
If it's a selectable item, we hide vpa and select that item. What do you think?